### ISSUE TITLE
Validator fails ES2019 public field declaration

### ISSUE THREAD

**Issue by erosman** [mozilla/addons-linter#3062]
ref: 75.0a1 (2020-02-27) (64-bit)

Validator fails ES2019 public field declaration even though Firefox works fine with.

![image](https://user-images.githubusercontent.com/6458876/75563215-38d38280-5a5f-11ea-94e0-2c1d1e8ee5f1.png)

Example:
```JavaScript
class MyClass {

  a = 1;
  b = 2;
  c = 3;

}
```

PS. If support was added to Firefox, which version is it from?

**Comment by Rob--W** [comment]
This is not a "variable outside the constructor". It is a (public) field declaration, supported in Firefox as of version 75 (and Chrome 74+), with the relevant bug being https://bugzilla.mozilla.org/show_bug.cgi?id=1499448 . The documentation is at https://developer.mozilla.org/en-US/docs/Web/JavaScript/Reference/Classes/Class_fields#Public_fields (scroll a bit below to "Public instance fields" for your specific case).

**Comment by erosman** [comment]
Thank you Rob. Title updated.

https://developer.mozilla.org/en-US/docs/Web/JavaScript/Reference/Classes/Class_fields#Public_fields  states support Firefox 69 !!

**Comment by Rob--W** [comment]
I stand corrected; Public instance fields (what you're using here) have already been supported since 69 (https://bugzilla.mozilla.org/show_bug.cgi?id=1555464).

I had public *static* fields in mind (syntax is `class {static name = value}`), which differs from what you've reported here, and is supported in Firefox 75+ (https://bugzilla.mozilla.org/show_bug.cgi?id=1535804).

**Comment by lfilho** [comment]
Arrived here for the same reason.
Tried running the `web-ext lint --ignore-files "**/.*js"` in an attempt to let the addons-linter only lint the manifes.json file and leave all the .js files to my local eslint setup, but that `--ignore-files` argument is actually applied to `web-ext` and not only to the `lint` subcommand, failing the lint command completely with: `"content script defined in the manifest could not be found at "content_script.js""`.

Right now I'm just stopping to use `web-ext lint` anymore :-/

Ideally addons-linter would be able to identify an eslint already installed in the user's package.json and use that one instead?

Or maybe a even better idea is instead of having a customer linter here, you would create an eslint plugin/preset for WebExtensions instead?

Anyway, thanks!

**Comment by lfilho** [comment]
Another great workaround would be if we could just say: `web-ext lint file-to-lint-1 file-to-lint-2`. This way i could just point it to manifes.json while keeping the rest of of my .js files being linted by the `eslint` in my project.

**Comment by willdurand** [comment]
see also: https://github.com/mozilla/addons-linter/issues/2758

**Comment by devsecdan** [comment]
> 
> 
> I stand corrected; Public instance fields (what you're using here) have already been supported since 69 (https://bugzilla.mozilla.org/show_bug.cgi?id=1555464).
> 
> I had public _static_ fields in mind (syntax is `class {static name = value}`), which differs from what you've reported here, and is supported in Firefox 75+ (https://bugzilla.mozilla.org/show_bug.cgi?id=1535804).

Ran into this issue today. Unable to publish my extension because it makes use of static fields.

**Comment by rpl** [comment]
Unfortunately this issue is still blocked on an upstream change: eslint officially supporting the class fields tc39 proposal.

It looks that the class fields tc39 proposal is still at stage 3 (see https://github.com/tc39/proposal-class-fields) and I'm pretty sure that eslint does not usually introduce support for new javascript syntaxes until their tc39 proposal does reach stage 4.

**Comment by regseb** [comment]
You can use `@babel/eslint-parser` and `@babel/plugin-proposal-class-properties` in [ESLint configuration](https://github.com/mozilla/addons-linter/blob/master/src/scanners/javascript.js#L88) to support class field.

```javascript
{
    parser: "@babel/eslint-parser",
    parserOptions: {
        requireConfigFile: false,
        babelOptions: {
            plugins: [
                "@babel/plugin-proposal-class-properties"
            ]
        }
    }
}
```

**Comment by muzuiget** [comment]
Please fix this issue.

Currently "Class Fields" feature is on stage 4, and Webpack 5 already supports it a few days ago.

**Comment by Rob--W** [comment]
Depends on https://github.com/eslint/eslint/issues/14343

**Comment by lydell** [comment]
[ESLint v8.0.0-beta.0](https://eslint.org/blog/2021/08/eslint-v8.0.0-beta.0-released) (and [ESLint v8.0.0-beta.1](https://eslint.org/blog/2021/08/eslint-v8.0.0-beta.1-released)) have been released – with support for class fields – so if anyone would like to begin work on this, I guess it’s possible now!

**Comment by ioanarusiczki** [comment]
@willdurand 
I verified this locally and with AMO dev using the example above and from [docs](https://developer.mozilla.org/en-US/docs/Web/JavaScript/Reference/Classes/Public_class_fields#Public_fields)

![dev](https://user-images.githubusercontent.com/33448286/140516308-f396fa81-5c5a-43eb-a8e4-7f6d783b3c66.png)

![linter 4 0 0](https://user-images.githubusercontent.com/33448286/140516330-75a08db8-c167-4610-bc82-58a9c4175113.png)

This was before:

![stage](https://user-images.githubusercontent.com/33448286/140516372-1885cfca-c185-4ef2-b42e-ad7484e6d1fb.png)

[publicfield.zip](https://github.com/mozilla/addons-linter/files/7486249/publicfield.zip)
[publicfield2.zip](https://github.com/mozilla/addons-linter/files/7486231/publicfield2.zip)
